diffusion/models/stable_diffusion.py

```diff
@@ -448,45 +448,44 @@ def _prepare_text_embeddings(self, prompt, tokenized_prompts, tokenized_pad_mask
         device = self.text_encoder.device
         pooled_text_embeddings = None
         if prompt_embeds is None:
-            max_length = None if self.sdxl else self.tokenizer.model_max_length
             if tokenized_prompts is None:
                 tokenized_out = self.tokenizer(prompt,
                                                padding='max_length',
-                                               max_length=max_length,
+                                               max_length=self.tokenizer.model_max_length,
                                                truncation=True,
                                                return_tensors='pt')
-                tokenized_prompts = tokenized_out.input_ids
-                if self.mask_pad_tokens:
-                    tokenized_pad_mask = tokenized_out.attention_mask
-                if self.sdxl:
-                    tokenized_prompts = torch.stack([tokenized_prompts[0], tokenized_prompts[1]], dim=1)
-                    if self.mask_pad_tokens:
-                        # For cross attention mask, take union of masks (want [B, 77])
-                        tokenized_pad_mask = torch.logical_or(tokenized_pad_mask[0], tokenized_pad_mask[1]).to(
-                            tokenized_pad_mask[0].dtype).to(device)
+                tokenized_prompts = tokenized_out['input_ids']
+                tokenized_pad_mask = tokenized_out['attention_mask']
+            if tokenized_pad_mask is not None:
+                tokenized_pad_mask = tokenized_pad_mask.to(device)
+            text_encoder_out = self.text_encoder(tokenized_prompts.to(device), attention_mask=tokenized_pad_mask)
+            prompt_embeds = text_encoder_out[0]
             if self.sdxl:
-                text_embeddings, pooled_text_embeddings = self.text_encoder(
-                    [tokenized_prompts[:, 0, :].to(device), tokenized_prompts[:, 1, :].to(device)])  # type: ignore
-            else:
-                text_embeddings = self.text_encoder(tokenized_prompts.to(device))[0]  # type: ignore
+                if len(text_encoder_out) <= 1:
+                    raise RuntimeError('SDXL requires text encoder output to include a pooled text embedding')
```

Comment:
```suggestion
                    raise ValueError('SDXL requires text encoder output to include a pooled text embedding')
```

Reply:
Should this be a ValueError? The error is from using a text encoder that doesn't output a pooled embedding